nipype/interfaces/base/traits_extension.py

```diff
@@ -322,15 +338,11 @@ def validate(self, object, name, value):
 
         newvalue = value
 
+        inner_trait = self.inner_traits()[0]
         if not isinstance(value, list) \
-            or (self.inner_traits() and
-                isinstance(self.inner_traits()[0].trait_type,
-                           traits.List) and not
-                isinstance(self.inner_traits()[0].trait_type,
-                           InputMultiPath) and
-                isinstance(value, list) and
-                value and not
-                isinstance(value[0], list)):
+            or (isinstance(inner_trait.trait_type, traits.List) and
+                not isinstance(inner_trait.trait_type, InputMultiPath) and
+                not isinstance(value[0], list)):
```

Comment:
Just to clarify:

* `self.inner_traits()` is always `True` for a `traits.List` (default value: `(traits.Any(),)`)
* `isinstance(value, list)` is always `True` at this point in the short-circuit
* `bool(value)` is always True, or we would have returned `Undefined` already

These changes aren't really related to the rest of the PR, but I thought I would be modifying code in here at first, so factoring the `inner_trait` local variable led to this.

Reply:
@effigies what if `value = []`?

Reply:
ok, sorry it would return undefined in l.331